### PR TITLE
Add more Colorize overloads and fix docs

### LINKED ISSUE #11828: Colorize docs are wrong
Take a look at the docs:

https://crystal-lang.org/api/1.3.2/Colorize.html

For example:

```crystal
require "colorize"

"foo".colorize(0, 255, 255) # => "foo" in aqua

# This is the same as:

"foo".colorize(Colorize::ColorRGB.new(0, 255, 255)) # => "foo" in aqua
```

^ None of the above work

```crystal
require "colorize"

"foo".colorize(Colorize::Color256.new(208)) # => "foo" in orange
```

Doesn't work!

The only thing that did work was:

```crystal
"foo".colorize.fore(Colorize::ColorRGB.new(0, 255, 255))
```

### PR DESCRIPTION
**Pull request by asterite**
Fixes #11828

### CODE REVIEW

File: src/colorize.cr [hunk 1]
```diff
@@ -169,8 +172,25 @@ module Colorize::ObjectExtensions
     Colorize::Object.new(self)
   end
 
-  # Turns `self` into a `Colorize::Object` and colors it with a color.
-  def colorize(fore)
+  # Turns `self` into a `Colorize::Object` and colors it with the given `Color256`
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply src/colorize.cr]: Sorry, I forgot to mention this applies to the following doc comments as well.

File: src/colorize.cr [hunk 2]
```diff
@@ -169,8 +172,25 @@ module Colorize::ObjectExtensions
...
 
-  # Turns `self` into a `Colorize::Object` and colors it with a color.
-  def colorize(fore)
+  # Wraps `self` in a `Colorize::Object` and colors it with the given `Color256`
+  # made up from the single *fore* byte.
+  def colorize(fore : UInt8)
+    Colorize::Object.new(self).fore(fore)
+  end
+
+  # Turns `self` into a `Colorize::Object` and colors it with the given `Color256` made
```
Comment (asterite) [inline src/colorize.cr:181]:
```suggestion
  # Wraps `self` in a `Colorize::Object` and colors it with the given `Color256` made
```

File: src/colorize.cr [hunk 3]
```diff
@@ -169,8 +172,25 @@ module Colorize::ObjectExtensions
...
+  # up from the given *r*ed, *g*reen and *b*lue values.
+  def colorize(r : UInt8, g : UInt8, b : UInt8)
+    Colorize::Object.new(self).fore(r, g, b)
+  end
+
+  # Turns `self` into a `Colorize::Object` and colors it with the given *fore* `Color`.
```
Comment (asterite) [inline src/colorize.cr:187]:
```suggestion
  # Wraps `self` in a `Colorize::Object` and colors it with the given *fore* `Color`.
```

File: src/colorize.cr [hunk 4]
```diff
@@ -169,8 +172,25 @@ module Colorize::ObjectExtensions
...
+  def colorize(fore : Color)
+    Colorize::Object.new(self).fore(fore)
+  end
+
+  # Turns `self` into a `Colorize::Object` and colors it with the given *fore* color.
```
Comment (asterite) [inline src/colorize.cr:192]:
```suggestion
  # Wraps `self` in a `Colorize::Object` and colors it with the given *fore* color.
```

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**